base/sharedarray.jl

```diff
@@ -2,7 +2,21 @@
 
 import .Serializer: serialize_cycle_header, serialize_type, writetag, UNDEFREF_TAG
 
+const SA_CTR = Ref(1)
+next_sa_ctr() = (id = SA_CTR[]; SA_CTR[] = id+1; id)
+
+mutable struct SA_ID
```

Comment:
You are right. A previous(different) implementation was relying on a finalizer to cleanup.

Now we can just use `Base.Distributed.RRID`. Will update.